che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/NativeGitConnection.java

```diff
@@ -323,12 +324,21 @@ public boolean isInsideWorkTree() throws GitException {
         final EmptyGitCommand emptyGitCommand = nativeGit.createEmptyGitCommand();
         
         // command "rev-parse --is-inside-work-tree" returns true/false
-        emptyGitCommand.setNextParameter("rev-parse")
-                       .setNextParameter("--is-inside-work-tree")
-                       .execute();
-                       
-        final String output = emptyGitCommand.getText();
-        return Boolean.valueOf(output);
+        try {
+            emptyGitCommand.setNextParameter("rev-parse")
+                           .setNextParameter("--is-inside-work-tree")
+                           .execute();
+                   
+            final String output = emptyGitCommand.getText();
+            return Boolean.valueOf(output);
+        } catch(GitException ge) {
+            String msg = ge.getMessage();
+            if (msg != null && notInGitRepoErrorPattern.matcher(msg).find()) {
```

Comment:
I meant method `matches()`
Here is part of javadocs of these methods:
 - find() `Attempts to find the next subsequence of the input sequence that matches the pattern.`
 - matches() `Attempts to match the entire region against the pattern.` 
And in this case we check if entire string matches.